### PR TITLE
rustdoc: Fix missing enum variant reexports

### LINKED ISSUE #35488: Missing documentation when re-exporting enumeration variants
When a private child module defines an enum and pub-uses its variants, and the parent module re-exports the content of the child module:

``` rust
mod foo {
    pub enum Foo {
        Bar,
    }
    pub use self::Foo::*;
}

pub use self::foo::*;
```

it is not visible in the documentation of the parent module that the variants of the enum are directly accessible to the user. However, if the child module was public, the pub-use of the variants would be shown on the pages of both modules.

Regards,
Ivan

### PR DESCRIPTION
**Pull request by ollie27**
Fixes #35488

### CODE REVIEW

File: src/librustdoc/visit_ast.rs [hunk 1]
```diff
@@ -329,25 +329,21 @@ impl<'a, 'tcx> RustdocVisitor<'a, 'tcx> {
...
-                if glob {
-                    match it.node {
-                        hir::ItemMod(ref m) => {
-                            for i in &m.item_ids {
-                                let i = self.cx.tcx.hir.expect_item(i.id);
-                                self.visit_item(i, None, om);
-                            }
-                        }
-                        hir::ItemEnum(..) => {}
-                        _ => { panic!("glob not mapped to a module or enum"); }
```
Comment (QuietMisdreavus) [inline src/librustdoc/visit_ast.rs]: It took me a couple reads to understand what was going on here, but just to straighten out my understanding: Reworking the match like this means that glob imports of enums will never be inlined, making them always appear as `pub use SomeEnum::*;`, whereas before they would be *treated* as they were inlined, but the variants would never be visited, leaving them completely undocumented? I haven't worked much with the AST crawling portion of rustdoc, but looking at the comments at the head of the function and the other change in `clean/inline.rs`, that's the impression I get from this change.

Comment (ollie27) [inline src/librustdoc/visit_ast.rs]: That's correct. In most cases already reexported enum variant appear as `pub use SomeEnum::*;` or `pub use SomeEnum::Foo;`, this PR fixes a couple of specific cases when they don't.

Comment (QuietMisdreavus) [inline src/librustdoc/visit_ast.rs]: Cool. I see you pushed an update, but r=me pending travis.

**Pull request merged**